scripts/sct_compute_mtr.py

```diff
@@ -56,52 +56,37 @@ def main(args=None):
 
     fname_mt0 = arguments['-mt0']
     fname_mt1 = arguments['-mt1']
-    remove_temp_files = int(arguments['-r'])
+    fname_mtr = arguments['-o']
+    file_out = os.path.basename(fname_mtr)[:os.path.basename(fname_mtr).rfind('.nii')]
+    ext_out = os.path.basename(fname_mtr)[os.path.basename(fname_mtr).rfind('.nii'):]
     verbose = int(arguments.get('-v'))
     sct.init_sct(log_level=verbose, update=True)  # Update log level
 
-    # Extract path/file/extension
-    path_mt0, file_mt0, ext_mt0 = sct.extract_fname(fname_mt0)
-    path_out, file_out, ext_out = '', file_out, ext_mt0
-
-    # create temporary folder
-    path_tmp = sct.tmp_create()
-
-    # Copying input data to tmp folder and convert to nii
-    sct.printv('\nCopying input data to tmp folder and convert to nii...', verbose)
-    from sct_convert import convert
-    convert(fname_mt0, os.path.join(path_tmp, "mt0.nii"), dtype=np.float32)
-    convert(fname_mt1, os.path.join(path_tmp, "mt1.nii"), dtype=np.float32)
-
-    # go to tmp folder
-    curdir = os.getcwd()
-    os.chdir(path_tmp)
-
     # compute MTR
     sct.printv('\nCompute MTR...', verbose)
-    nii_mt1 = msct_image.Image('mt1.nii')
+    nii_mt1 = msct_image.Image(fname_mt1)
     data_mt1 = nii_mt1.data
-    data_mt0 = msct_image.Image('mt0.nii').data
+    data_mt0 = msct_image.Image(fname_mt0).data
     data_mtr = 100 * (data_mt0 - data_mt1) / data_mt0
     # save MTR file
     nii_mtr = nii_mt1
     nii_mtr.data = data_mtr
-    nii_mtr.save("mtr.nii")
-    # sct.run(fsloutput+'fslmaths -dt double mt0.nii -sub mt1.nii -mul 100 -div mt0.nii -thr 0 -uthr 100 mtr.nii', verbose)
+    nii_mtr.save(fname_mtr)
+    # sct.run(fsloutput+'fslmaths -dt double mt0.nii -sub mt1.nii -mul 100 -div mt0.nii -thr 0 -uthr 100 file_out.nii', verbose)
 
-    # come back
-    os.chdir(curdir)
+    # if changing output file name or location, create folder with mt0.nii.gz and mt1.nii.gz files at precised location
+    if os.path.dirname(fname_mtr) != '.':
+        sct.copy(os.path.join('.', fname_mt0), os.path.dirname(fname_mtr))
+        sct.copy(os.path.join('.', fname_mt1), os.path.dirname(fname_mtr))
+
+    # go to output file directory
+    os.chdir(os.path.dirname(fname_mtr))
 
     # Generate output files
     sct.printv('\nGenerate output files...', verbose)
-    sct.generate_output_file(os.path.join(path_tmp, "mtr.nii"), os.path.join(path_out, file_out + ext_out))
-
-    # Remove temporary files
-    if remove_temp_files == 1:
-        sct.printv('\nRemove temporary files...')
-        sct.rmtree(path_tmp)
+    sct.generate_output_file(os.path.join('.', file_out + ext_out), '.')
 
-    sct.display_viewer_syntax([fname_mt0, fname_mt1, file_out])
+    sct.display_viewer_syntax([fname_mt0, fname_mt1, file_out + ext_out])
```

Comment:
just use fname_mtr

Reply:
Understood, changes in commit [25defd6](https://github.com/YangHee-Min/spinalcordtoolbox/commit/25defd618e92c464100eee81949aafddf4f8cbfd)